service/rds/rdsutils/connect.go

```diff
@@ -33,7 +34,7 @@ import (
 //
 // See http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.html
 // for more information on using IAM database authentication with RDS.
-func BuildAuthToken(endpoint, region, dbUser string, credProvider aws.CredentialsProvider) (string, error) {
+func BuildAuthToken(ctx context.Context, endpoint, region, dbUser string, credProvider aws.CredentialsProvider) (string, error) {
```

Comment:
suggest instead of taking credential provide this utility takes a signer instead.

```go
HTTPV4Signer interface {
    Presign(context.Context, *http.Request, ...)
}
```